File: src/api/dotnet/SetSort.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    SetSort.cs

Abstract:

    Z3 Managed API: Set Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// Set sorts.
    /// </summary>
    public class SetSort : Sort
    {
        #region Internal
        internal SetSort(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        internal SetSort(Context ctx, Sort ty)
            : base(ctx, Native.Z3_mk_set_sort(ctx.nCtx, ty.NativeObject))
        {
            Debug.Assert(ctx != null);
            Debug.Assert(ty != null);
        }
        #endregion
    }
}

File: src/api/java/SetSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    SetSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/ 

package com.microsoft.z3;

/**
 * Set sorts.
 **/
public class SetSort<D extends Sort> extends ArraySort<D, BoolSort>
{
    SetSort(Context ctx, long obj)
    {
        super(ctx, obj);
    }

    SetSort(Context ctx, D ty)
    {
        super(ctx, Native.mkSetSort(ctx.nCtx(), ty.getNativeObject()));
    }
}
